adi/ad7091rx.py

```diff
@@ -0,0 +1,151 @@
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = [
+            "ad7091r-2",
+            "ad7091r-4",
+            "ad7091r-8",
+        ]
+
+        self._ctrl = None
+
+        if not device_name:
+            device_name = compatible_parts[2]
+        else:
+            if device_name not in compatible_parts:
+                raise Exception("Not a compatible device: " + device_name)
+
+        # Selecting the device matching device_name AD7091RX family as working device.
+        for device in self._ctx.devices:
+            if device.name == device_name:
+                self._ctrl = device
+                self._rxadc = device
+                break
+
+        for ch in self._ctrl.channels:
+            name = ch._id
+            self._rx_channel_names.append(name)
+            self.channel.append(self._channel(self._ctrl, name))
+            setattr(self, name, self._channel(self._ctrl, name))
+
+        rx.__init__(self)
+
+    class _channel(attribute):
+        """AD7091R-8/-4/-2 Input Voltage Channels"""
+
+        def __init__(self, ctrl, channel_name):
+            self.name = channel_name
+            self._ctrl = ctrl
+
+        @property
+        def raw(self):
+            """AD7091r channel raw value"""
+            return self._get_iio_attr(self.name, "raw", False)
+
+        @property
+        def scale(self):
+            """AD7091r channel scale"""
+            return float(self._get_iio_attr_str(self.name, "scale", False))
+
+        @property
+        def thresh_falling_value(self):
+            """Get channel threshold falling value"""
+            return self._get_iio_attr(self.name, "thresh_falling_value", False)
+
+        @thresh_falling_value.setter
+        def thresh_falling_value(self, value):
+            """Set channel threshold falling value"""
+            self._set_iio_attr(
+                self.name, "thresh_falling_value", False, str(Decimal(value))
+            )
+
+        @property
+        def thresh_rising_value(self):
+            """Get channel threshold rising value"""
+            return self._get_iio_attr(self.name, "thresh_rising_value", False)
+
+        @thresh_rising_value.setter
+        def thresh_rising_value(self, value):
+            """Set channel threshold rising value"""
+            self._set_iio_attr(
+                self.name, "thresh_rising_value", False, str(Decimal(value))
+            )
+
+        @property
+        def thresh_either_hysteresis(self):
+            """Get channel threshold either hysteresis value"""
+            return self._get_iio_attr(self.name, "thresh_either_hysteresis", False)
+
+        @thresh_either_hysteresis.setter
+        def thresh_either_hysteresis(self, value):
+            """Set channel threshold either hysteresis value"""
+            self._set_iio_attr(
+                self.name, "thresh_either_hysteresis", False, str(Decimal(value))
+            )
```

Comment:
Hey @SaikiranGudla, it will be great to have threshold support at some point but libiio doesn't look like it's already supported. I see in libiio code there is `enum iio_event_type` in include/iio/iio.h and I even installed libiio v1 from main branch but got `KeyError: 'thresh_rising_value'` at `ad7091r_dev.channel[0].thresh_rising_value = 0x10F`.
My suggestion is to drop threshold support for now.